modules/apps/object/object-service/src/main/java/com/liferay/object/internal/scope/DepotObjectScopeProviderImpl.java

```diff
@@ -52,6 +53,23 @@ public String[] getRootPanelCategoryKeys() {
 		return new String[0];
 	}
 
+	@Override
+	public Long getValidGroupId(long companyId, String scopeKey) {
+		Group group = _groupLocalService.fetchGroup(companyId, scopeKey);
+
+		if (group == null) {
+			group = _groupLocalService.fetchGroup(GetterUtil.getLong(scopeKey));
+		}
+
+		if (isValidGroupId(group.getGroupId()) ||
+			isValidGroupId(group.getLiveGroupId())) {
```

Comment:
@devpedrotavares when calling `isValidGroupId` twice here, we will make two calls to `fetchGroup` again.. 

This seems unnecessary as we already have the `group` here.

I would try something like:

```
@Override
public Long getValidGroupId(long companyId, String scopeKey) {
	Group group = _groupLocalService.fetchGroup(companyId, scopeKey);

	if (group == null) {
		group = _groupLocalService.fetchGroup(GetterUtil.getLong(scopeKey));
	}

	if (_isValidGroupId(group)) {
		return group.getGroupId();
	}

	if(_isValidGroupId(_groupLocalService.fetchGroup(group.getLiveGroupId()))) {
		....
	}

	return null;
}


@Override
public boolean isValidGroupId(long groupId) {
	return _isValidGroupId(_groupLocalService.fetchGroup(groupId));
}

private boolean _isValidGroupId(Group group) {
	if ((group != null) && ....) {
		return true;
	}

	return false;
}
```

Reply:
Sure, I changed it